File: lib/features/employee/new_recruite/domain/recruit_info.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:police_com/core/enums/all_enums.dart';

import 'recruitment_announcement.dart';

part 'recruit_info.freezed.dart';
part 'recruit_info.g.dart';

@freezed
abstract class RecruitInfo with _$RecruitInfo {
  const factory RecruitInfo({
    required String? recruitId,
    required String? announcementId,
    @JsonKey(includeFromJson: false, includeToJson: false)
    RecruitmentAnnouncement? announcement,
    required String? firstName,
    required String? fatherName,
    String? grandName,
    required Gender gender,
    required DateTime birthDate,
    String? motherName,
    required String? mobile,
    String? email,
    @Default(Country.ethiopia) Country nationality,
    @Default(NewRecruitStatus.pending) NewRecruitStatus recruitStatus,
  }) = _RecruitInfo;

  factory RecruitInfo.fromJson(Map<String, dynamic> json) =>
      _$RecruitInfoFromJson(json);
}

File: Backend_dot_net_c_sharp_models_and_eums_to_convert_to_dart /Models/Employee/NewRecruit/RecruitInfo.cs
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
using PeerageHRIS.Enumeration;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace PeerageHRIS.Models.Employee.NewRecruit
{
    public class RecruitInfo
    {
        [Key]
        [Required]
        [Display(Name ="Recruit Id")]
        public string? RecruitId { get; set; }

        [Required]
        [Display(Name = "Announcement")]
        public string? AnnouncementId { get; set; }

        [ForeignKey(nameof(AnnouncementId))]
        [JsonIgnore]
        [ValidateNever]
        public RecruitmentAnnouncement? Announcement { get; set; }

        [Required]
        [Display(Name = "First Name")]
        public string? FirstName { get; set; }

        [Required]
        [Display(Name = "Father Name")]
        public string? FatherName { get; set; }

        [Display(Name = "Grand Father Name")]
        public string? GrandName { get; set; }

        [Required]
        [Display(Name = "Gender")]
        public Gender Gender { get; set; }

        [Required]
        [Display(Name = "Birth Date")]
        [DataType(DataType.Date)]
        public DateTime BirthDate { get; set; }

        [Display(Name = "Mother Full Name")]
        public string? MotherName { get; set; }

        [Required]
        [Display(Name = "Mobile Number")]
        [DataType(DataType.PhoneNumber)]
        public string? Mobile { get; set; }

        [Display(Name = "Email")]
        public string? Email { get; set; }

        [Display(Name = "Nationality")]
        public Country Nationality { get; set; } = Country.Ethiopia;

        // Computed property for dropdown display
        [NotMapped]
        [Display(Name ="Full Name")]
        public string FullName => $"{FirstName} {FatherName} {GrandName}";

    }
}
